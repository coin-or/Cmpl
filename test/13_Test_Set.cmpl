
const:
	# Definition Vektoren aus Sets
	s[] := (
			..,									# unendlicher Set: alle integers und alle strings
			..6,								# unendlicher Set: alle integers kleiner gleich dem Wert
			6..,								# unendlicher Set: alle integers groesser gleich dem Wert
			.integer.,							# unendlicher Set: alle integers
			.string.,							# unendlicher Set: alle strings
			set(),								# leerer Set
			0..6,								# Algorithmischer Set mit Schrittweite 1
			1..7,								# Algorithmischer Set mit Schrittweite 1
			4..9,								# Algorithmischer Set mit Schrittweite 1
			3(-1)2,								# Algorithmischer Set mit beliebiger Schrittweite
			set( "qwe" ),						# Aufzaehlungsset
			set( 7, "qwe", 6, "fe", 5, 8 )		# Aufzaehlungsset
		);

	t[] := (
			1..8,
			5..7,
			7(-1)5,
			set( 7, "qwe", 6, "fe", 5, 8 ),
			set( "qwe" ),
			set( 6, 7, 5 ),
			set( 5, 6, 7, 8, "fe", "qwe" )
		);

	# Ausgabe Sets
	{ i in 1..dim(s[]): echo i + " (" + type(s[i]) + "): " + string(s[i]); }

	# Set-Operationen
	{ i in defset(s[5..5]), j in defset(s[]):
		echo string(s[i]) + " + " + string(s[j]) + " : " + string(s[i] + s[j]);		# Vereinigung
		echo string(s[i]) + " * " + string(s[j]) + " : " + string(s[i] * s[j]);		# Durchschnitt
	}

	# Vereinigung von Sets ueber SUM
	q := sum{ i in 12(-1)6: s[i] };
	echo q;
	{ i in q: echo i; }

	# Test ob Element in Set enthalten
	{ i in 1..dim(s[]): echo s[i];
		echo "1: " + (1 element s[i]);
		echo "8: " + (8 element s[i]);
		echo "fe: " + ("fe" element s[i]);
	}

	# Vergleich von Sets
	{ i in defset(t[]), j in defset(t[]):
		echo string(t[i]) + " < " + string(t[j]) + " : " + (t[i] < t[j]);		# keine Beruecksichtigung der Reihenfolge
		echo string(t[i]) + " <= " + string(t[j]) + " : " + (t[i] <= t[j]);		# keine Beruecksichtigung der Reihenfolge
		echo string(t[i]) + " > " + string(t[j]) + " : " + (t[i] > t[j]);		# keine Beruecksichtigung der Reihenfolge
		echo string(t[i]) + " >= " + string(t[j]) + " : " + (t[i] >= t[j]);		# keine Beruecksichtigung der Reihenfolge
		echo string(t[i]) + " = " + string(t[j]) + " : " + (t[i] = t[j]);		# keine Beruecksichtigung der Reihenfolge
		echo string(t[i]) + " == " + string(t[j]) + " : " + (t[i] == t[j]);		# auch Reihenfolge der Elemente wird geprueft
		echo string(t[i]) + " <> " + string(t[j]) + " : " + (t[i] <> t[j]);		# keine Beruecksichtigung der Reihenfolge
		echo string(t[i]) + " != " + string(t[j]) + " : " + (t[i] != t[j]);		# auch Reihenfolge der Elemente wird geprueft
	}

	# Ueber Bedingung konstruierte Sets
	a := set{ i in t[7], !(i element t[6]): i };
	echo a;

	b := set{ i in t[7], !(i element t[5]): i };
	echo b;

