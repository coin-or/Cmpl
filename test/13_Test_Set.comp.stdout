1 (set (infinite all)): ..
2 (set (infinite with upper bound)): ..6
3 (set (infinite with lower bound)): 6..
4 (set (infinite all integers)): .integer.
5 (set (infinite all string)): .string.
6 (set (empty)): set()
7 (set (interval from 0)): 0..6
8 (set (interval from 1)): 1..7
9 (set (algorithmic)): 4..9
10 (set (algorithmic)): 3(-1)2
11 (set (enum)): set( "qwe" )
12 (set (enum)): set( 7, "qwe", 6, "fe", 5, 8 )
.string. + .. : ..
.string. * .. : .string.
.string. + ..6 : .string.
.string. * ..6 : set()
.string. + 6.. : .string.
.string. * 6.. : set()
.string. + .integer. : ..
.string. * .integer. : set()
.string. + .string. : .string.
.string. * .string. : .string.
.string. + set() : .string.
.string. * set() : set()
.string. + 0..6 : .string.
.string. * 0..6 : set()
.string. + 1..7 : .string.
.string. * 1..7 : set()
.string. + 4..9 : .string.
.string. * 4..9 : set()
.string. + 3(-1)2 : .string.
.string. * 3(-1)2 : set()
.string. + set( "qwe" ) : .string.
.string. * set( "qwe" ) : set( "qwe" )
.string. + set( 7, "qwe", 6, "fe", 5, 8 ) : .string.
.string. * set( 7, "qwe", 6, "fe", 5, 8 ) : set( "fe", "qwe" )
set( 7, "qwe", 6, "fe", 5, 8, 3, 2, 4, 9, 1, 0 )
7
qwe
6
fe
5
8
3
2
4
9
1
0
..
1: 1
8: 1
fe: 1
..6
1: 1
8: 0
fe: 0
6..
1: 0
8: 1
fe: 0
.integer.
1: 1
8: 1
fe: 0
.string.
1: 0
8: 0
fe: 1
set()
1: 0
8: 0
fe: 0
0..6
1: 1
8: 0
fe: 0
1..7
1: 1
8: 0
fe: 0
4..9
1: 0
8: 1
fe: 0
3(-1)2
1: 0
8: 0
fe: 0
set( "qwe" )
1: 0
8: 0
fe: 0
set( 7, "qwe", 6, "fe", 5, 8 )
1: 0
8: 1
fe: 1
1..8 < 1..8 : 0
1..8 <= 1..8 : 1
1..8 > 1..8 : 0
1..8 >= 1..8 : 1
1..8 = 1..8 : 1
1..8 == 1..8 : 1
1..8 <> 1..8 : 0
1..8 != 1..8 : 0
1..8 < 5..7 : 0
1..8 <= 5..7 : 0
1..8 > 5..7 : 1
1..8 >= 5..7 : 1
1..8 = 5..7 : 0
1..8 == 5..7 : 0
1..8 <> 5..7 : 1
1..8 != 5..7 : 1
1..8 < 7(-1)5 : 0
1..8 <= 7(-1)5 : 0
1..8 > 7(-1)5 : 1
1..8 >= 7(-1)5 : 1
1..8 = 7(-1)5 : 0
1..8 == 7(-1)5 : 0
1..8 <> 7(-1)5 : 1
1..8 != 7(-1)5 : 1
1..8 < set( 7, "qwe", 6, "fe", 5, 8 ) : 0
1..8 <= set( 7, "qwe", 6, "fe", 5, 8 ) : 0
1..8 > set( 7, "qwe", 6, "fe", 5, 8 ) : 0
1..8 >= set( 7, "qwe", 6, "fe", 5, 8 ) : 0
1..8 = set( 7, "qwe", 6, "fe", 5, 8 ) : 0
1..8 == set( 7, "qwe", 6, "fe", 5, 8 ) : 0
1..8 <> set( 7, "qwe", 6, "fe", 5, 8 ) : 1
1..8 != set( 7, "qwe", 6, "fe", 5, 8 ) : 1
1..8 < set( "qwe" ) : 0
1..8 <= set( "qwe" ) : 0
1..8 > set( "qwe" ) : 0
1..8 >= set( "qwe" ) : 0
1..8 = set( "qwe" ) : 0
1..8 == set( "qwe" ) : 0
1..8 <> set( "qwe" ) : 1
1..8 != set( "qwe" ) : 1
1..8 < set( 6, 7, 5 ) : 0
1..8 <= set( 6, 7, 5 ) : 0
1..8 > set( 6, 7, 5 ) : 1
1..8 >= set( 6, 7, 5 ) : 1
1..8 = set( 6, 7, 5 ) : 0
1..8 == set( 6, 7, 5 ) : 0
1..8 <> set( 6, 7, 5 ) : 1
1..8 != set( 6, 7, 5 ) : 1
1..8 < set( 5, 6, 7, 8, "fe", "qwe" ) : 0
1..8 <= set( 5, 6, 7, 8, "fe", "qwe" ) : 0
1..8 > set( 5, 6, 7, 8, "fe", "qwe" ) : 0
1..8 >= set( 5, 6, 7, 8, "fe", "qwe" ) : 0
1..8 = set( 5, 6, 7, 8, "fe", "qwe" ) : 0
1..8 == set( 5, 6, 7, 8, "fe", "qwe" ) : 0
1..8 <> set( 5, 6, 7, 8, "fe", "qwe" ) : 1
1..8 != set( 5, 6, 7, 8, "fe", "qwe" ) : 1
5..7 < 1..8 : 1
5..7 <= 1..8 : 1
5..7 > 1..8 : 0
5..7 >= 1..8 : 0
5..7 = 1..8 : 0
5..7 == 1..8 : 0
5..7 <> 1..8 : 1
5..7 != 1..8 : 1
5..7 < 5..7 : 0
5..7 <= 5..7 : 1
5..7 > 5..7 : 0
5..7 >= 5..7 : 1
5..7 = 5..7 : 1
5..7 == 5..7 : 1
5..7 <> 5..7 : 0
5..7 != 5..7 : 0
5..7 < 7(-1)5 : 0
5..7 <= 7(-1)5 : 1
5..7 > 7(-1)5 : 0
5..7 >= 7(-1)5 : 1
5..7 = 7(-1)5 : 1
5..7 == 7(-1)5 : 0
5..7 <> 7(-1)5 : 0
5..7 != 7(-1)5 : 1
5..7 < set( 7, "qwe", 6, "fe", 5, 8 ) : 1
5..7 <= set( 7, "qwe", 6, "fe", 5, 8 ) : 1
5..7 > set( 7, "qwe", 6, "fe", 5, 8 ) : 0
5..7 >= set( 7, "qwe", 6, "fe", 5, 8 ) : 0
5..7 = set( 7, "qwe", 6, "fe", 5, 8 ) : 0
5..7 == set( 7, "qwe", 6, "fe", 5, 8 ) : 0
5..7 <> set( 7, "qwe", 6, "fe", 5, 8 ) : 1
5..7 != set( 7, "qwe", 6, "fe", 5, 8 ) : 1
5..7 < set( "qwe" ) : 0
5..7 <= set( "qwe" ) : 0
5..7 > set( "qwe" ) : 0
5..7 >= set( "qwe" ) : 0
5..7 = set( "qwe" ) : 0
5..7 == set( "qwe" ) : 0
5..7 <> set( "qwe" ) : 1
5..7 != set( "qwe" ) : 1
5..7 < set( 6, 7, 5 ) : 0
5..7 <= set( 6, 7, 5 ) : 1
5..7 > set( 6, 7, 5 ) : 0
5..7 >= set( 6, 7, 5 ) : 1
5..7 = set( 6, 7, 5 ) : 1
5..7 == set( 6, 7, 5 ) : 0
5..7 <> set( 6, 7, 5 ) : 0
5..7 != set( 6, 7, 5 ) : 1
5..7 < set( 5, 6, 7, 8, "fe", "qwe" ) : 1
5..7 <= set( 5, 6, 7, 8, "fe", "qwe" ) : 1
5..7 > set( 5, 6, 7, 8, "fe", "qwe" ) : 0
5..7 >= set( 5, 6, 7, 8, "fe", "qwe" ) : 0
5..7 = set( 5, 6, 7, 8, "fe", "qwe" ) : 0
5..7 == set( 5, 6, 7, 8, "fe", "qwe" ) : 0
5..7 <> set( 5, 6, 7, 8, "fe", "qwe" ) : 1
5..7 != set( 5, 6, 7, 8, "fe", "qwe" ) : 1
7(-1)5 < 1..8 : 1
7(-1)5 <= 1..8 : 1
7(-1)5 > 1..8 : 0
7(-1)5 >= 1..8 : 0
7(-1)5 = 1..8 : 0
7(-1)5 == 1..8 : 0
7(-1)5 <> 1..8 : 1
7(-1)5 != 1..8 : 1
7(-1)5 < 5..7 : 0
7(-1)5 <= 5..7 : 1
7(-1)5 > 5..7 : 0
7(-1)5 >= 5..7 : 1
7(-1)5 = 5..7 : 1
7(-1)5 == 5..7 : 0
7(-1)5 <> 5..7 : 0
7(-1)5 != 5..7 : 1
7(-1)5 < 7(-1)5 : 0
7(-1)5 <= 7(-1)5 : 1
7(-1)5 > 7(-1)5 : 0
7(-1)5 >= 7(-1)5 : 1
7(-1)5 = 7(-1)5 : 1
7(-1)5 == 7(-1)5 : 1
7(-1)5 <> 7(-1)5 : 0
7(-1)5 != 7(-1)5 : 0
7(-1)5 < set( 7, "qwe", 6, "fe", 5, 8 ) : 1
7(-1)5 <= set( 7, "qwe", 6, "fe", 5, 8 ) : 1
7(-1)5 > set( 7, "qwe", 6, "fe", 5, 8 ) : 0
7(-1)5 >= set( 7, "qwe", 6, "fe", 5, 8 ) : 0
7(-1)5 = set( 7, "qwe", 6, "fe", 5, 8 ) : 0
7(-1)5 == set( 7, "qwe", 6, "fe", 5, 8 ) : 0
7(-1)5 <> set( 7, "qwe", 6, "fe", 5, 8 ) : 1
7(-1)5 != set( 7, "qwe", 6, "fe", 5, 8 ) : 1
7(-1)5 < set( "qwe" ) : 0
7(-1)5 <= set( "qwe" ) : 0
7(-1)5 > set( "qwe" ) : 0
7(-1)5 >= set( "qwe" ) : 0
7(-1)5 = set( "qwe" ) : 0
7(-1)5 == set( "qwe" ) : 0
7(-1)5 <> set( "qwe" ) : 1
7(-1)5 != set( "qwe" ) : 1
7(-1)5 < set( 6, 7, 5 ) : 0
7(-1)5 <= set( 6, 7, 5 ) : 1
7(-1)5 > set( 6, 7, 5 ) : 0
7(-1)5 >= set( 6, 7, 5 ) : 1
7(-1)5 = set( 6, 7, 5 ) : 1
7(-1)5 == set( 6, 7, 5 ) : 0
7(-1)5 <> set( 6, 7, 5 ) : 0
7(-1)5 != set( 6, 7, 5 ) : 1
7(-1)5 < set( 5, 6, 7, 8, "fe", "qwe" ) : 1
7(-1)5 <= set( 5, 6, 7, 8, "fe", "qwe" ) : 1
7(-1)5 > set( 5, 6, 7, 8, "fe", "qwe" ) : 0
7(-1)5 >= set( 5, 6, 7, 8, "fe", "qwe" ) : 0
7(-1)5 = set( 5, 6, 7, 8, "fe", "qwe" ) : 0
7(-1)5 == set( 5, 6, 7, 8, "fe", "qwe" ) : 0
7(-1)5 <> set( 5, 6, 7, 8, "fe", "qwe" ) : 1
7(-1)5 != set( 5, 6, 7, 8, "fe", "qwe" ) : 1
set( 7, "qwe", 6, "fe", 5, 8 ) < 1..8 : 0
set( 7, "qwe", 6, "fe", 5, 8 ) <= 1..8 : 0
set( 7, "qwe", 6, "fe", 5, 8 ) > 1..8 : 0
set( 7, "qwe", 6, "fe", 5, 8 ) >= 1..8 : 0
set( 7, "qwe", 6, "fe", 5, 8 ) = 1..8 : 0
set( 7, "qwe", 6, "fe", 5, 8 ) == 1..8 : 0
set( 7, "qwe", 6, "fe", 5, 8 ) <> 1..8 : 1
set( 7, "qwe", 6, "fe", 5, 8 ) != 1..8 : 1
set( 7, "qwe", 6, "fe", 5, 8 ) < 5..7 : 0
set( 7, "qwe", 6, "fe", 5, 8 ) <= 5..7 : 0
set( 7, "qwe", 6, "fe", 5, 8 ) > 5..7 : 1
set( 7, "qwe", 6, "fe", 5, 8 ) >= 5..7 : 1
set( 7, "qwe", 6, "fe", 5, 8 ) = 5..7 : 0
set( 7, "qwe", 6, "fe", 5, 8 ) == 5..7 : 0
set( 7, "qwe", 6, "fe", 5, 8 ) <> 5..7 : 1
set( 7, "qwe", 6, "fe", 5, 8 ) != 5..7 : 1
set( 7, "qwe", 6, "fe", 5, 8 ) < 7(-1)5 : 0
set( 7, "qwe", 6, "fe", 5, 8 ) <= 7(-1)5 : 0
set( 7, "qwe", 6, "fe", 5, 8 ) > 7(-1)5 : 1
set( 7, "qwe", 6, "fe", 5, 8 ) >= 7(-1)5 : 1
set( 7, "qwe", 6, "fe", 5, 8 ) = 7(-1)5 : 0
set( 7, "qwe", 6, "fe", 5, 8 ) == 7(-1)5 : 0
set( 7, "qwe", 6, "fe", 5, 8 ) <> 7(-1)5 : 1
set( 7, "qwe", 6, "fe", 5, 8 ) != 7(-1)5 : 1
set( 7, "qwe", 6, "fe", 5, 8 ) < set( 7, "qwe", 6, "fe", 5, 8 ) : 0
set( 7, "qwe", 6, "fe", 5, 8 ) <= set( 7, "qwe", 6, "fe", 5, 8 ) : 1
set( 7, "qwe", 6, "fe", 5, 8 ) > set( 7, "qwe", 6, "fe", 5, 8 ) : 0
set( 7, "qwe", 6, "fe", 5, 8 ) >= set( 7, "qwe", 6, "fe", 5, 8 ) : 1
set( 7, "qwe", 6, "fe", 5, 8 ) = set( 7, "qwe", 6, "fe", 5, 8 ) : 1
set( 7, "qwe", 6, "fe", 5, 8 ) == set( 7, "qwe", 6, "fe", 5, 8 ) : 1
set( 7, "qwe", 6, "fe", 5, 8 ) <> set( 7, "qwe", 6, "fe", 5, 8 ) : 0
set( 7, "qwe", 6, "fe", 5, 8 ) != set( 7, "qwe", 6, "fe", 5, 8 ) : 0
set( 7, "qwe", 6, "fe", 5, 8 ) < set( "qwe" ) : 0
set( 7, "qwe", 6, "fe", 5, 8 ) <= set( "qwe" ) : 0
set( 7, "qwe", 6, "fe", 5, 8 ) > set( "qwe" ) : 1
set( 7, "qwe", 6, "fe", 5, 8 ) >= set( "qwe" ) : 1
set( 7, "qwe", 6, "fe", 5, 8 ) = set( "qwe" ) : 0
set( 7, "qwe", 6, "fe", 5, 8 ) == set( "qwe" ) : 0
set( 7, "qwe", 6, "fe", 5, 8 ) <> set( "qwe" ) : 1
set( 7, "qwe", 6, "fe", 5, 8 ) != set( "qwe" ) : 1
set( 7, "qwe", 6, "fe", 5, 8 ) < set( 6, 7, 5 ) : 0
set( 7, "qwe", 6, "fe", 5, 8 ) <= set( 6, 7, 5 ) : 0
set( 7, "qwe", 6, "fe", 5, 8 ) > set( 6, 7, 5 ) : 1
set( 7, "qwe", 6, "fe", 5, 8 ) >= set( 6, 7, 5 ) : 1
set( 7, "qwe", 6, "fe", 5, 8 ) = set( 6, 7, 5 ) : 0
set( 7, "qwe", 6, "fe", 5, 8 ) == set( 6, 7, 5 ) : 0
set( 7, "qwe", 6, "fe", 5, 8 ) <> set( 6, 7, 5 ) : 1
set( 7, "qwe", 6, "fe", 5, 8 ) != set( 6, 7, 5 ) : 1
set( 7, "qwe", 6, "fe", 5, 8 ) < set( 5, 6, 7, 8, "fe", "qwe" ) : 0
set( 7, "qwe", 6, "fe", 5, 8 ) <= set( 5, 6, 7, 8, "fe", "qwe" ) : 1
set( 7, "qwe", 6, "fe", 5, 8 ) > set( 5, 6, 7, 8, "fe", "qwe" ) : 0
set( 7, "qwe", 6, "fe", 5, 8 ) >= set( 5, 6, 7, 8, "fe", "qwe" ) : 1
set( 7, "qwe", 6, "fe", 5, 8 ) = set( 5, 6, 7, 8, "fe", "qwe" ) : 1
set( 7, "qwe", 6, "fe", 5, 8 ) == set( 5, 6, 7, 8, "fe", "qwe" ) : 0
set( 7, "qwe", 6, "fe", 5, 8 ) <> set( 5, 6, 7, 8, "fe", "qwe" ) : 0
set( 7, "qwe", 6, "fe", 5, 8 ) != set( 5, 6, 7, 8, "fe", "qwe" ) : 1
set( "qwe" ) < 1..8 : 0
set( "qwe" ) <= 1..8 : 0
set( "qwe" ) > 1..8 : 0
set( "qwe" ) >= 1..8 : 0
set( "qwe" ) = 1..8 : 0
set( "qwe" ) == 1..8 : 0
set( "qwe" ) <> 1..8 : 1
set( "qwe" ) != 1..8 : 1
set( "qwe" ) < 5..7 : 0
set( "qwe" ) <= 5..7 : 0
set( "qwe" ) > 5..7 : 0
set( "qwe" ) >= 5..7 : 0
set( "qwe" ) = 5..7 : 0
set( "qwe" ) == 5..7 : 0
set( "qwe" ) <> 5..7 : 1
set( "qwe" ) != 5..7 : 1
set( "qwe" ) < 7(-1)5 : 0
set( "qwe" ) <= 7(-1)5 : 0
set( "qwe" ) > 7(-1)5 : 0
set( "qwe" ) >= 7(-1)5 : 0
set( "qwe" ) = 7(-1)5 : 0
set( "qwe" ) == 7(-1)5 : 0
set( "qwe" ) <> 7(-1)5 : 1
set( "qwe" ) != 7(-1)5 : 1
set( "qwe" ) < set( 7, "qwe", 6, "fe", 5, 8 ) : 1
set( "qwe" ) <= set( 7, "qwe", 6, "fe", 5, 8 ) : 1
set( "qwe" ) > set( 7, "qwe", 6, "fe", 5, 8 ) : 0
set( "qwe" ) >= set( 7, "qwe", 6, "fe", 5, 8 ) : 0
set( "qwe" ) = set( 7, "qwe", 6, "fe", 5, 8 ) : 0
set( "qwe" ) == set( 7, "qwe", 6, "fe", 5, 8 ) : 0
set( "qwe" ) <> set( 7, "qwe", 6, "fe", 5, 8 ) : 1
set( "qwe" ) != set( 7, "qwe", 6, "fe", 5, 8 ) : 1
set( "qwe" ) < set( "qwe" ) : 0
set( "qwe" ) <= set( "qwe" ) : 1
set( "qwe" ) > set( "qwe" ) : 0
set( "qwe" ) >= set( "qwe" ) : 1
set( "qwe" ) = set( "qwe" ) : 1
set( "qwe" ) == set( "qwe" ) : 1
set( "qwe" ) <> set( "qwe" ) : 0
set( "qwe" ) != set( "qwe" ) : 0
set( "qwe" ) < set( 6, 7, 5 ) : 0
set( "qwe" ) <= set( 6, 7, 5 ) : 0
set( "qwe" ) > set( 6, 7, 5 ) : 0
set( "qwe" ) >= set( 6, 7, 5 ) : 0
set( "qwe" ) = set( 6, 7, 5 ) : 0
set( "qwe" ) == set( 6, 7, 5 ) : 0
set( "qwe" ) <> set( 6, 7, 5 ) : 1
set( "qwe" ) != set( 6, 7, 5 ) : 1
set( "qwe" ) < set( 5, 6, 7, 8, "fe", "qwe" ) : 1
set( "qwe" ) <= set( 5, 6, 7, 8, "fe", "qwe" ) : 1
set( "qwe" ) > set( 5, 6, 7, 8, "fe", "qwe" ) : 0
set( "qwe" ) >= set( 5, 6, 7, 8, "fe", "qwe" ) : 0
set( "qwe" ) = set( 5, 6, 7, 8, "fe", "qwe" ) : 0
set( "qwe" ) == set( 5, 6, 7, 8, "fe", "qwe" ) : 0
set( "qwe" ) <> set( 5, 6, 7, 8, "fe", "qwe" ) : 1
set( "qwe" ) != set( 5, 6, 7, 8, "fe", "qwe" ) : 1
set( 6, 7, 5 ) < 1..8 : 1
set( 6, 7, 5 ) <= 1..8 : 1
set( 6, 7, 5 ) > 1..8 : 0
set( 6, 7, 5 ) >= 1..8 : 0
set( 6, 7, 5 ) = 1..8 : 0
set( 6, 7, 5 ) == 1..8 : 0
set( 6, 7, 5 ) <> 1..8 : 1
set( 6, 7, 5 ) != 1..8 : 1
set( 6, 7, 5 ) < 5..7 : 0
set( 6, 7, 5 ) <= 5..7 : 1
set( 6, 7, 5 ) > 5..7 : 0
set( 6, 7, 5 ) >= 5..7 : 1
set( 6, 7, 5 ) = 5..7 : 1
set( 6, 7, 5 ) == 5..7 : 0
set( 6, 7, 5 ) <> 5..7 : 0
set( 6, 7, 5 ) != 5..7 : 1
set( 6, 7, 5 ) < 7(-1)5 : 0
set( 6, 7, 5 ) <= 7(-1)5 : 1
set( 6, 7, 5 ) > 7(-1)5 : 0
set( 6, 7, 5 ) >= 7(-1)5 : 1
set( 6, 7, 5 ) = 7(-1)5 : 1
set( 6, 7, 5 ) == 7(-1)5 : 0
set( 6, 7, 5 ) <> 7(-1)5 : 0
set( 6, 7, 5 ) != 7(-1)5 : 1
set( 6, 7, 5 ) < set( 7, "qwe", 6, "fe", 5, 8 ) : 1
set( 6, 7, 5 ) <= set( 7, "qwe", 6, "fe", 5, 8 ) : 1
set( 6, 7, 5 ) > set( 7, "qwe", 6, "fe", 5, 8 ) : 0
set( 6, 7, 5 ) >= set( 7, "qwe", 6, "fe", 5, 8 ) : 0
set( 6, 7, 5 ) = set( 7, "qwe", 6, "fe", 5, 8 ) : 0
set( 6, 7, 5 ) == set( 7, "qwe", 6, "fe", 5, 8 ) : 0
set( 6, 7, 5 ) <> set( 7, "qwe", 6, "fe", 5, 8 ) : 1
set( 6, 7, 5 ) != set( 7, "qwe", 6, "fe", 5, 8 ) : 1
set( 6, 7, 5 ) < set( "qwe" ) : 0
set( 6, 7, 5 ) <= set( "qwe" ) : 0
set( 6, 7, 5 ) > set( "qwe" ) : 0
set( 6, 7, 5 ) >= set( "qwe" ) : 0
set( 6, 7, 5 ) = set( "qwe" ) : 0
set( 6, 7, 5 ) == set( "qwe" ) : 0
set( 6, 7, 5 ) <> set( "qwe" ) : 1
set( 6, 7, 5 ) != set( "qwe" ) : 1
set( 6, 7, 5 ) < set( 6, 7, 5 ) : 0
set( 6, 7, 5 ) <= set( 6, 7, 5 ) : 1
set( 6, 7, 5 ) > set( 6, 7, 5 ) : 0
set( 6, 7, 5 ) >= set( 6, 7, 5 ) : 1
set( 6, 7, 5 ) = set( 6, 7, 5 ) : 1
set( 6, 7, 5 ) == set( 6, 7, 5 ) : 1
set( 6, 7, 5 ) <> set( 6, 7, 5 ) : 0
set( 6, 7, 5 ) != set( 6, 7, 5 ) : 0
set( 6, 7, 5 ) < set( 5, 6, 7, 8, "fe", "qwe" ) : 1
set( 6, 7, 5 ) <= set( 5, 6, 7, 8, "fe", "qwe" ) : 1
set( 6, 7, 5 ) > set( 5, 6, 7, 8, "fe", "qwe" ) : 0
set( 6, 7, 5 ) >= set( 5, 6, 7, 8, "fe", "qwe" ) : 0
set( 6, 7, 5 ) = set( 5, 6, 7, 8, "fe", "qwe" ) : 0
set( 6, 7, 5 ) == set( 5, 6, 7, 8, "fe", "qwe" ) : 0
set( 6, 7, 5 ) <> set( 5, 6, 7, 8, "fe", "qwe" ) : 1
set( 6, 7, 5 ) != set( 5, 6, 7, 8, "fe", "qwe" ) : 1
set( 5, 6, 7, 8, "fe", "qwe" ) < 1..8 : 0
set( 5, 6, 7, 8, "fe", "qwe" ) <= 1..8 : 0
set( 5, 6, 7, 8, "fe", "qwe" ) > 1..8 : 0
set( 5, 6, 7, 8, "fe", "qwe" ) >= 1..8 : 0
set( 5, 6, 7, 8, "fe", "qwe" ) = 1..8 : 0
set( 5, 6, 7, 8, "fe", "qwe" ) == 1..8 : 0
set( 5, 6, 7, 8, "fe", "qwe" ) <> 1..8 : 1
set( 5, 6, 7, 8, "fe", "qwe" ) != 1..8 : 1
set( 5, 6, 7, 8, "fe", "qwe" ) < 5..7 : 0
set( 5, 6, 7, 8, "fe", "qwe" ) <= 5..7 : 0
set( 5, 6, 7, 8, "fe", "qwe" ) > 5..7 : 1
set( 5, 6, 7, 8, "fe", "qwe" ) >= 5..7 : 1
set( 5, 6, 7, 8, "fe", "qwe" ) = 5..7 : 0
set( 5, 6, 7, 8, "fe", "qwe" ) == 5..7 : 0
set( 5, 6, 7, 8, "fe", "qwe" ) <> 5..7 : 1
set( 5, 6, 7, 8, "fe", "qwe" ) != 5..7 : 1
set( 5, 6, 7, 8, "fe", "qwe" ) < 7(-1)5 : 0
set( 5, 6, 7, 8, "fe", "qwe" ) <= 7(-1)5 : 0
set( 5, 6, 7, 8, "fe", "qwe" ) > 7(-1)5 : 1
set( 5, 6, 7, 8, "fe", "qwe" ) >= 7(-1)5 : 1
set( 5, 6, 7, 8, "fe", "qwe" ) = 7(-1)5 : 0
set( 5, 6, 7, 8, "fe", "qwe" ) == 7(-1)5 : 0
set( 5, 6, 7, 8, "fe", "qwe" ) <> 7(-1)5 : 1
set( 5, 6, 7, 8, "fe", "qwe" ) != 7(-1)5 : 1
set( 5, 6, 7, 8, "fe", "qwe" ) < set( 7, "qwe", 6, "fe", 5, 8 ) : 0
set( 5, 6, 7, 8, "fe", "qwe" ) <= set( 7, "qwe", 6, "fe", 5, 8 ) : 1
set( 5, 6, 7, 8, "fe", "qwe" ) > set( 7, "qwe", 6, "fe", 5, 8 ) : 0
set( 5, 6, 7, 8, "fe", "qwe" ) >= set( 7, "qwe", 6, "fe", 5, 8 ) : 1
set( 5, 6, 7, 8, "fe", "qwe" ) = set( 7, "qwe", 6, "fe", 5, 8 ) : 1
set( 5, 6, 7, 8, "fe", "qwe" ) == set( 7, "qwe", 6, "fe", 5, 8 ) : 0
set( 5, 6, 7, 8, "fe", "qwe" ) <> set( 7, "qwe", 6, "fe", 5, 8 ) : 0
set( 5, 6, 7, 8, "fe", "qwe" ) != set( 7, "qwe", 6, "fe", 5, 8 ) : 1
set( 5, 6, 7, 8, "fe", "qwe" ) < set( "qwe" ) : 0
set( 5, 6, 7, 8, "fe", "qwe" ) <= set( "qwe" ) : 0
set( 5, 6, 7, 8, "fe", "qwe" ) > set( "qwe" ) : 1
set( 5, 6, 7, 8, "fe", "qwe" ) >= set( "qwe" ) : 1
set( 5, 6, 7, 8, "fe", "qwe" ) = set( "qwe" ) : 0
set( 5, 6, 7, 8, "fe", "qwe" ) == set( "qwe" ) : 0
set( 5, 6, 7, 8, "fe", "qwe" ) <> set( "qwe" ) : 1
set( 5, 6, 7, 8, "fe", "qwe" ) != set( "qwe" ) : 1
set( 5, 6, 7, 8, "fe", "qwe" ) < set( 6, 7, 5 ) : 0
set( 5, 6, 7, 8, "fe", "qwe" ) <= set( 6, 7, 5 ) : 0
set( 5, 6, 7, 8, "fe", "qwe" ) > set( 6, 7, 5 ) : 1
set( 5, 6, 7, 8, "fe", "qwe" ) >= set( 6, 7, 5 ) : 1
set( 5, 6, 7, 8, "fe", "qwe" ) = set( 6, 7, 5 ) : 0
set( 5, 6, 7, 8, "fe", "qwe" ) == set( 6, 7, 5 ) : 0
set( 5, 6, 7, 8, "fe", "qwe" ) <> set( 6, 7, 5 ) : 1
set( 5, 6, 7, 8, "fe", "qwe" ) != set( 6, 7, 5 ) : 1
set( 5, 6, 7, 8, "fe", "qwe" ) < set( 5, 6, 7, 8, "fe", "qwe" ) : 0
set( 5, 6, 7, 8, "fe", "qwe" ) <= set( 5, 6, 7, 8, "fe", "qwe" ) : 1
set( 5, 6, 7, 8, "fe", "qwe" ) > set( 5, 6, 7, 8, "fe", "qwe" ) : 0
set( 5, 6, 7, 8, "fe", "qwe" ) >= set( 5, 6, 7, 8, "fe", "qwe" ) : 1
set( 5, 6, 7, 8, "fe", "qwe" ) = set( 5, 6, 7, 8, "fe", "qwe" ) : 1
set( 5, 6, 7, 8, "fe", "qwe" ) == set( 5, 6, 7, 8, "fe", "qwe" ) : 1
set( 5, 6, 7, 8, "fe", "qwe" ) <> set( 5, 6, 7, 8, "fe", "qwe" ) : 0
set( 5, 6, 7, 8, "fe", "qwe" ) != set( 5, 6, 7, 8, "fe", "qwe" ) : 0
set( 8, "fe", "qwe" )
set( 5, 6, 7, 8, "fe" )
